Signal/src/views/AudioProgressView.swift

```diff
@@ -0,0 +1,97 @@
+//
+//  Copyright (c) 2017 Open Whisper Systems. All rights reserved.
+//
+
+import UIKit
+
+@objc class AudioProgressView: UIView {
+
+    override var bounds: CGRect {
+        didSet {
+            if oldValue != bounds {
+                updateSubviews()
+            }
+        }
+    }
+
+    override var frame: CGRect {
+        didSet {
+            if oldValue != frame {
+                updateSubviews()
+            }
+        }
+    }
+
+    var horizontalBarColor = UIColor.black {
+        didSet {
+            updateContent()
+        }
+    }
+
+    var progressColor = UIColor.blue {
+        didSet {
+            updateContent()
+        }
+    }
+
+    private let horizontalBarLayer: CAShapeLayer
+    private let progressLayer: CAShapeLayer
+
+    var progress: CGFloat = 0 {
+        didSet {
+            if oldValue != progress {
+                updateContent()
+            }
+        }
+    }
+
+    @available(*, unavailable, message:"use delegate: constructor instead.")
```

Comment:
nit: unavailable comment seems like it should be like "use init() constructor instead"